networkx/algorithms/traversal/breadth_first_search.py

```diff
@@ -310,8 +314,9 @@ def bfs_successors(G, source, depth_limit=None, sort_neighbors=None):
     Returns
     -------
     succ: iterator
-       (node, successors) iterator where successors is the list of
-       successors of the node.
+       (node, successors) iterator where `successors` is the list of
+       successors of `node`, `node` is a transitive successor of `source`
+       and `successors` is non-empty.
```

Comment:
Would it be better to define transitive successor here? Also, the first time I read this I couldn't tell if it is a list of 3 traits that successors has or if the latter phrases are simply describing other true statements. When I got to `successors is non-empty`, I decided it must be 3 ideas about successors.  I suggest splitting the sentence:

```suggestion
       (node, successors) iterator where `successors` is the non-empty list of
       successors of `node` in a breadth first search from `source`.
       To appear in the iterator,  `node` must have successors.
```

Reply:
Thanks! Splitting the sentence makes it more readable.

I think, we should not go into (direct) successors vs. transitive successors here. If (at all), I feel the natural place would be in the doctest of DiGraph.successors (maybe that could be another pull request).